mechanics/CREATING-A-SANDBOX-REPO.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Creating a sandbox repo"
+linkTitle: "Sandbox repo process"
+weight: 30
+type: "docs"
+---
+
+# Background
+
+The [`knative-sandbox`](https://github.com/knative-sandbox) GitHub org exists to
+hold "non-core" Knative components which are owned and sponsored by a Working
+Group but are not part of an official Knative release.
+
+## Criteria
+
+A Working Group Lead (either
+[technical](../ROLES.md#working-group-technical-lead) or
+[execution](../ROLES.md#working-group-execution-lead)) may request a new repo in
+`knative-sandbox` from any [TOC member](../TECH-OVERSIGHT-COMMITTEE.md).
+Generally, the request will be granted, though the TOC may have additional
+questions or suggest an alternate mechanism in some cases.
+
+Some of the following steps may require permissions that are only available to
+the TOC or Steering Committee, though others are largely self-service or require
+other WGs to review and approve impacting changes.
+
+### Technical requirements
+
+- Any contributed code should be contributed by the original authors or
+  copyright holders under an Apache 2.0 license.
+
+- Names between `knative-sandbox` and the main `knative` repo should not
+  overlap. This simplifies promoting repos between the two orgs and setting up
+  `knative.dev` import paths for golang.
+
+- Prow automation for tests is not necessarily required for `knative-sandbox`,
+  but the Google CLA bot and OWNERS files/tide merge should be enforced.
```

Comment:
How do you set up the CLA bot?

Reply:
So, Google owns all sandbox projects?

Reply:
Can we switch to a DCO for these?

Reply:
I think you'll need to take that to the TOC -- Google owns the trademark on "Knative", and the current requirements from #23 are to have the Google CLA-bot enforced.

Reply:
The CLA-bot is set up on the org as a whole.

I think Ville discovered that you can't select which checks are enforced until they've been run on at least one PR, so I moved the branch-protection setup to an additional item after creating a test PR for Prow.

Reply:
So just to be clear.... if someone puts a project under sandbox but then later on decides that since Knative "core" doesn't want it, can they take it and run with it? I guess they can "fork" their own stuff, but will they have to choose a new name? Can they remove it w/o Google's approval?

Reply:
The CLA doesn't say Google owns the code - it gives Google license to use it.

Reply:
The details are in the CLA which is a pretty long read. The Istio team wrote this very short summary about what it means: "...is necessary because you own the copyright to your changes, even after your contribution becomes part of this project. So this agreement simply gives Google permission to use and redistribute your contributions as part of the project." https://github.com/istio/community/blob/master/CONTRIBUTING.md#contributor-license-agreements

Reply:
Doug, can you add the trademark question to the overall trademark issue: https://github.com/knative/community/issues/9

Reply:
Can we just switch to DCO for everything?

Reply:
No, Google projects require the Google CLA.

Reply:
The Google CLA has language like `You hereby grant to Google and to recipients of software distributed by Google....`  does this imply that "Google is distributing Knative" and not the community? Does it mean Google and the community are one in the same?